vtex-sites/gatsby.store
# chore: add 'optionalDependencies' to package.json

## What's the purpose of this pull request?
This PR adds the `optionalDependencies` section at `package.json` and adds dependencies that are not necessary to build the application to it.
This was made so the build time is reduced, since we are now skipping unnecessary build dependencies such as `eslint` and `storybook`. A faster yarn install can now be performed with `yarn install --ignore-optional`

Benchmarks (at my local development machine):
| Command                                                 | Output            | Note                                                                                                    |
|---------------------------------------------------------|-------------------|---------------------------------------------------------------------------------------------------------|
| `rm -rf node_modules && yarn install`                   | ✨ Done in 82.41s. | This was the command we were running before when building the application. All packages are installed.  |
| `rm -rf node_modules && yarn install --ignore-optional` | ✨ Done in 48.51s. | This is the new command that we intend to use in future webops builds, ignoring packages in `optionalDependencies` |

We can observe a 41.13% reduction in the execution of `yarn install` when using the `--ignore-optional` flag


## How does it work?
By using the `optionalDependencies` category, we can now only install packages that we are really going to use during build time.

## How to test it?
Basically, replicate the benchmarks mentioned in the first section. Here's a guide:
- Clear node_modules
- Run `yarn install`, with and without the `--ignore-optional` flag
- Verify that the package install time decreases when using the flag
- Run `yarn build`
- Verify that the build completes successfully without the optional packages
